docs/proposals/20201020-capi-provider-operator.md

```diff
@@ -0,0 +1,1097 @@
+- `FetchConfiguration` is used only during init and upgrade operations.
+  Changes made to the contents of `FetchConfiguration` will not trigger a
+  reconciliation. This is similar behavior to `ProviderSpec.SecretName`.
+
+**Validation Rules for ProviderSpec.ManagerSpec**
+- The ControllerManagerConfigurationSpec is a type from
+  `controller-runtime/pkg/config` and is an embedded into the `ManagerSpec`.
+  This type will expose LeaderElection, SyncPeriod, Webhook, Health and
+  Metrics configurations.
+- If `ManagerSpec.Debug` is set to true, the operator will not allow changes
+  to other properties since it is in Debug mode.
+- If you need to set specific concurrency values for each reconcile loop (e.g.
+  `awscluster-concurrency`), you can leave
+  `ManagerSpec.MaxConcurrentReconciles` nil and use `Container.Args`.
+- If `ManagerSpec.MaxConcurrentReconciles` is set and a specific concurrency
+  flag such as `awscluster-concurrency` is set on the `Container.Args`, then
+  the more specific concurrency flag will have higher precedence.
+
+
+**Validation Rules for ContainerSpec**
+- The `ContainerSpec.Args` will ignore the key `namespace` since the operator
+  enforces a deployment model where all the providers should be configured to
+  watch all the namespaces.
+- Explicit controller manager properties defined in the `Provider.ManagerSpec`
+  will have higher precedence than those defined in `ContainerSpec.Args`. That
+  is, if `ManagerSpec.SyncPeriod` is defined it will be used instead of the
+  container arg `sync-period`. This is true also for
+  `ManagerSpec.FeatureGates`, that is, it will have higher precedence to the
+  container arg `feature-gates`.
+- If no `ContainerSpec.Resources` are defined, the defaults on the Deployment
+  object within the provider’s components yaml will be used.
+
+
+**Open Questions**
+1. Should the CRDs be global objects or namespaced objects?
+   A provider lives in a namespace (e.g. capa-system), so it is nice to have
+   the object namespaced and use metadata.namespace. However, with the new
+   approach to multitenancy, we should enforce are like singleton (a
+   provider, identified by Kind & name, should be unique at cluster level)
+   It will be nice to provide a nice UX around kubectl get <providers>
+
+   _We will scope the CRDs to be namespaced. This will allow us to enforce
+   RBAC restrictions if needed. This also allows us to install multiple
+   versions of the controllers (grouped within namespaces) in the same
+   management cluster although this scenario will not be supported natively in
+   the v1alpha4 iteration._
+
+1. What should the Group/Version for this API be?
+   Currently, we have it as `management.cluster.x-k8s.io/v1alpha1`. Should
+   the version be `v1alpha4` to match the CAPI version it is being introduced
+   in? This would be consistent with what we do for experiments.
+
+   _We can determine this during implementation phase. But I think it makes
```

Comment:
Given that this point is deferred to the implementation phase, I suggest dropping the question